src/main/java/practice/StreamPractice.java

```diff
@@ -14,18 +40,28 @@
      * "Can't get min value from list: method_input_list"
      */
     public int findMinEvenNumber(List<String> numbers) {
-        return 0;
+        return numbers.stream()
+                .flatMap(s -> Arrays.stream(s.split(",")))
+                .mapToInt(Integer::parseInt)
+                .filter(i -> i % 2 == 0)
+                .min()
+                .orElseThrow(() -> new RuntimeException("Can't get min value from list"));
     }
-
+    
     /**
      * Given a List of Integer numbers,
      * subtract 1 from each element on an odd position (having the odd index).
      * Then return the average of all odd numbers or throw NoSuchElementException.
      */
     public Double getOddNumsAverage(List<Integer> numbers) {
-        return 0D;
+        return IntStream.range(0, numbers.size())
+                .map(i -> i % 2 != 0 ? numbers.get(i) - 1 : numbers.get(i))
+                .filter(i -> i % 2 != 0)
+                .mapToDouble(i -> i)
+                .average()
+                .orElseThrow(NoSuchElementException::new);
```

Comment:
```suggestion
                .average()
                .get()
```
get will thro NoSuchElement exception by default

Reply:
humm.... I just thought that we want to define it explicitly, sure why not)

Reply:
that's predictable behavior that's why there is no need to explicitly declare it